### PR TITLE
refactor: move height caluclation

### PR DESCRIPTION
**Pull request by haxxmaxx**

### CODE REVIEW

**Review by a-m-dev — APPROVED** [review]

File: src/pivot-table/hooks/__tests__/use-grid-height.test.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  LeftDimensionData,
+  PageInfo,
+  Rect,
+  StyleService,
+  TopDimensionData,
+} from "../../../types/types";
+import * as styleProvider from "../../contexts/StyleProvider";
+import useGridHeight from "../use-grid-height";
+
+describe("useColumnWidth", () => {
```
Comment (cbt1) [inline src/pivot-table/hooks/__tests__/use-grid-height.test.ts:15]:
```suggestion
describe("useGridHeight", () => {
```

File: src/pivot-table/hooks/use-grid-height.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  const containerHeight = totalDataHeight + headerGridHeight;
+
+  // Top grid should always have height to support cases when there is no top data but it need to occupy space to currecly render headers
+  const topGridHeight = headerCellHeight * Math.max(topDimensionData.rowCount, 1);
+  const leftGridHeight = Math.min(tableRect.height - headerGridHeight - GRID_BORDER, totalDataHeight);
+  const dataGridHeight = Math.min(tableRect.height - topGridHeight - GRID_BORDER, totalDataHeight);
+
+  const rowsCanFitInTableViewPort = Math.floor(tableRect.height / contentCellHeight);
+  const rowsInCurrentPage = Object.values(Object.values(leftDimensionData.grid).at(-1) || {}).length;
+  const showLastBottomBorder = rowsInCurrentPage < rowsCanFitInTableViewPort;
```
Comment (cbt1) [inline src/pivot-table/hooks/use-grid-height.ts:33]:
```suggestion
  const showLastBottomBorder = pageInfo.rowsOnCurrentPage < rowsCanFitInTableViewPort;
```

**Review by cbt1 — APPROVED** [review]
LGTM, only minor tweaks

File: src/pivot-table/hooks/use-grid-height.ts (hunk 2 above)
Comment (haxxmaxx) [reply src/pivot-table/hooks/use-grid-height.ts:33]: so we shouldn't use the row count thing on the line above?

Comment (cbt1) [reply src/pivot-table/hooks/use-grid-height.ts:33]: No pageInfo already have that value

Comment (cbt1) [reply src/pivot-table/hooks/use-grid-height.ts:33]: WAIT!

Comment (cbt1) [reply src/pivot-table/hooks/use-grid-height.ts:33]: Let me double check this.

Comment (cbt1) [reply src/pivot-table/hooks/use-grid-height.ts:33]: `const rowsInCurrentPage = Object.values(Object.values(leftDimensionData.grid).at(-1) || {}).length;` is actually "number of that have been fetched". Which is working here but not really want we want.

Please go ahead and use `pageInfo.rowsOnCurrentPage`.

Comment (haxxmaxx) [reply src/pivot-table/hooks/use-grid-height.ts:33]: nice, that simplifies things

**Pull request merged**